python/tank/bootstrap/manager.py

```diff
@@ -506,8 +506,10 @@ def prepare_engine(self, engine_name, entity):
         :param entity: An entity link. If the entity is not a project, the project for that entity will be resolved.
         :type project: Dictionary with keys ``type`` and ``id``, or ``None`` for the site
 
-        :returns: Path to the pipeline configuration.
-        :rtype: str
+        :returns: Path to the fully realized pipeline configuration on disk and to the descriptor that
+            spawned it.
+
+        :rtype: (str, :class:`sgtk.descriptor.ConfigDescriptor`)
```

Comment:
Ah, should have mentionned this in the code review. This was introduced in the zero config branch. As such, it is not part of the public API.